apps/web/src/services/dashboards/shared/dashboard-widget-input-form/dashboard-widget-options-form/DashboardWidgetOptionDropdown.vue

```diff
@@ -2,129 +2,311 @@
 }>();
-const emit = defineEmits<{(e: 'update:inherit', isInherit: boolean): void;
-    (e: 'update:selected', selected: SelectDropdownMenuItem[]): void;
+const emit = defineEmits<{(e: 'update:is-valid', isValid: boolean): void;
     (e: 'delete'): void;
 }>();
 
+const widgetFormStore = useWidgetFormStore();
+const widgetFormState = widgetFormStore.$state;
 const state = reactive({
-    showErrorMessage: false,
+    // property schema
+    schemaProperty: computed<WidgetOptionsSchemaProperty|undefined>(() => widgetFormStore.widgetConfig?.options_schema?.properties?.[props.propertyName]),
+    label: computed(() => state.schemaProperty?.name ?? props.propertyName),
+    selectionType: computed<WidgetOptionsSchemaProperty['selection_type']>(() => state.schemaProperty?.selection_type),
+    inherit: computed<boolean>(() => !!widgetFormState.inheritOptions?.[props.propertyName]?.enabled),
+    inheritanceMode: computed<InheritanceMode>(() => state.schemaProperty?.inheritance_mode),
+    //
+    selected: [] as SelectDropdownMenuItem[],
     errorMessage: computed<string|undefined>(() => {
-        if (props.inherit) {
-            // TODO: implement inherit case
-            // const errorText = i18n.t('DASHBOARDS.WIDGET.VALIDATION_PROPERTY_NOT_EXIST') as string;
-            return undefined;
+        if (state.schemaProperty?.optional) return undefined;
+        if (!state.selected?.length) {
+            return i18n.t('DASHBOARDS.WIDGET.NO_SELECTED_ITEM') as string;
         }
+        return undefined;
+    }),
+});
+const menuState = reactive({
+    variableModels: computed<IBaseVariableModel[]>(() => getVariableModels(state.schemaProperty)),
+    inheritOptionMenuHandler: computed(() => getInheritOptionMenuHandler(state.schemaProperty)),
+    menuHandlers: computed(() => {
+        if (!state.schemaProperty) return [];
 
-        if (state.showErrorMessage && !props.selected?.length) {
-            return i18n.t('DASHBOARDS.WIDGET.NO_SELECTED_ITEM') as string;
+        if (widgetFormState.inheritOptions?.[props.propertyName]?.enabled) {
+            return [menuState.inheritOptionMenuHandler];
         }
 
-        return undefined;
+        // get options from schema dependencies
+        const options = {}; // e.g. { data_source_id: 'ds-1' }
+        Object.entries(state.schemaProperty.dependencies ?? {})?.forEach(([optionName, reference]) => {
+            options[reference.reference_key] = widgetFormState.widgetOptions[optionName];
+        });
+
+        return menuState.variableModels.map((variableModel) => getVariableModelMenuHandler(variableModel, options));
     }),
 });
 
-const handleUpdateVisibleMenu = (visible: boolean) => {
-    if (state.showErrorMessage || !visible) return;
-    state.showErrorMessage = true;
+/* Util */
+const getVariableModels = (schema?: WidgetOptionsSchemaProperty): IBaseVariableModel[] => {
+    if (!schema?.item_options) return [];
+
+    const variableModels: IBaseVariableModel[] = [];
```

Comment:
suggestion:
```suggestion
    const variableModels: VariableModel[] = [];
```
we can use class as interface :)